```
DEBUG - test_bgp_99_bmp (__main__.TestProtocolsBGP.test_bgp_99_bmp) ... FAIL
DEBUG - 
DEBUG - ======================================================================
DEBUG - FAIL: test_bgp_99_bmp (__main__.TestProtocolsBGP.test_bgp_99_bmp)
DEBUG - ----------------------------------------------------------------------
DEBUG - Traceback (most recent call last):
DEBUG -   File "/usr/libexec/vyos/tests/smoke/cli/test_protocols_bgp.py", line 1918, in test_bgp_99_bmp
DEBUG -     self.assertNotIn('source-interface', frrconfig)
DEBUG - AssertionError: 'source-interface' unexpectedly found in 'router bgp 64512\n no bgp ebgp-requires-policy\n no bgp reject-as-sets\n no bgp default ipv4-unicast\n no bgp network import-check\n !\n bmp mirror buffer-limit 32000000\n !\n bmp targets instance-bmp\n  bmp mirror\n  bmp monitor ipv4 unicast pre-policy\n  bmp monitor ipv4 unicast loc-rib\n  bmp monitor ipv6 unicast pre-policy\n  bmp monitor ipv6 unicast loc-rib\n  bmp connect 127.0.0.1 port 5000 min-retry 1024 max-retry 2048 source-interface eth0\n exit\nexit'
DEBUG - 
DEBUG - ----------------------------------------------------------------------
DEBUG - Ran 35 tests in 330.661s
DEBUG - 
DEBUG - FAILED (failures=1)
```
The test will succeed after the FRR fix [patch](https://github.com/vyos/vyos-build/pull/1225) is merged

--- a/data/templates/frr/bgpd.frr.j2
+++ b/data/templates/frr/bgpd.frr.j2
@@ -538,7 +538,7 @@ router bgp {{ system_as }} {{ 'vrf ' ~ vrf if vrf is vyos_defined }} {{ 'as-nota
 {%                 endif %}
 {%             endif %}
 {%             if bmp_config.address is vyos_defined %}
-  bmp connect {{ bmp_config.address }} port {{ bmp_config.port }} min-retry {{ bmp_config.min_retry }} max-retry {{ bmp_config.max_retry }}
+  bmp connect {{ bmp_config.address }} port {{ bmp_config.port }} min-retry {{ bmp_config.min_retry }} max-retry {{ bmp_config.max_retry }} {{ 'source-interface ' ~ bmp_config.source_interface if bmp_config.source_interface is vyos_defined }}
 {%             endif %}
 {%         endfor %}
  exit

--- a/interface-definitions/include/bgp/protocol-common-config.xml.i
+++ b/interface-definitions/include/bgp/protocol-common-config.xml.i
@@ -874,6 +874,7 @@
       <children>
         #include <include/address-ipv4-ipv6-single.xml.i>
         #include <include/port-number.xml.i>
+        #include <include/source-interface.xml.i>
         <leafNode name="port">
           <defaultValue>5000</defaultValue>
         </leafNode>

--- a/smoketest/scripts/cli/test_protocols_bgp.py
+++ b/smoketest/scripts/cli/test_protocols_bgp.py
@@ -1852,6 +1852,7 @@ class TestProtocolsBGP(VyOSUnitTestSHIM.TestCase):
         mirror_buffer = '32000000'
         bmp_path = base_path + ['bmp']
         target_path = bmp_path + ['target', target_name]
+        source_iface = 'eth0'
 
         # by default the 'bmp' module not loaded for the bgpd expect Error
         self.cli_set(bmp_path)
@@ -1879,6 +1880,7 @@ class TestProtocolsBGP(VyOSUnitTestSHIM.TestCase):
 
         # config other bmp options
         self.cli_set(target_path + ['address', target_address])
+        self.cli_set(target_path + ['source-interface', source_iface])
         self.cli_set(bmp_path + ['mirror-buffer-limit', mirror_buffer])
         self.cli_set(target_path + ['port', target_port])
         self.cli_set(target_path + ['min-retry', min_retry])
@@ -1899,7 +1901,21 @@ class TestProtocolsBGP(VyOSUnitTestSHIM.TestCase):
         self.assertIn(f'bmp monitor ipv6 unicast {monitor_ipv6}', frrconfig)
         self.assertIn(f'bmp monitor ipv4 unicast loc-rib', frrconfig)
         self.assertIn(f'bmp monitor ipv6 unicast loc-rib', frrconfig)
-        self.assertIn(f'bmp connect {target_address} port {target_port} min-retry {min_retry} max-retry {max_retry}', frrconfig)
+        self.assertIn(
+            f'bmp connect {target_address} port {target_port} min-retry {min_retry} max-retry {max_retry} source-interface {source_iface}',
+            frrconfig,
+        )
+
+        # verify source-interface is removed from FRR config after deletion
+        self.cli_delete(target_path + ['source-interface'])
+        self.cli_commit()
+
+        frrconfig = self.getFRRconfig(f'router bgp {ASN}', stop_section='^exit')
+        self.assertIn(
+            f'bmp connect {target_address} port {target_port} min-retry {min_retry} max-retry {max_retry}',
+            frrconfig,
+        )
+        self.assertNotIn('source-interface', frrconfig)
 
     def test_bgp_100_link_state(self):
         router_id = '127.0.0.1'

